Library/Homebrew/service.rb

```diff
@@ -371,7 +391,7 @@ def to_plist
       base = {
         Label:            @formula.plist_name,
         ProgramArguments: command,
-        RunAtLoad:        @run_type == RUN_TYPE_IMMEDIATE,
+        RunAtLoad:        @run_type == RUN_TYPE_IMMEDIATE || @run_at_load == true,
```

Comment:
This will also need changes in `brew services` to apply to linux

Reply:
AFAIK, on linux they start on load. Where did you guys handled this behavior? So that I can add this check there.

Reply:
It's in https://github.com/Homebrew/homebrew-services/blob/cb8908eaf4ec3caf595b66cd5bd6c6511d63a431/lib/service/services_cli.rb#L252-L255

Reply:
What if we keep it simple and constrain this parameter to only `run_type` of `interval`? That way we won't have to modify the `brew services`. Please check the newest commit.

Reply:
Yes we do, if you set this to false linux won't follow that directive.

Reply:
Isn't that the default behaviour? `OnActiveSec=0s` should in theory work on linux when this flag is set.

Reply:
But Linux doesn't use LaunchD. It doesn't care about the plist file. It cares about the commands in the file I linked above.

Reply:
Yes, I'm talking about this: https://github.com/Homebrew/brew/blob/16b51d2a431cfbe51ec63f6de979e087117ec975/Library/Homebrew/service.rb#L489

Reply:
I'm just saying that if the default behaviour on Linux is to not run the exec at the load time then the solution should be aimed towards running the exec on service load. Hence `OnActiveSec`. If it doesn't make sense then maybe I'm missing something here

Reply:
I'm not sure how `OnActiveSec` would work if you run `systemctl start <service>`.

Reply:
Afaik, that gets fired after the timer is activated and the `OnUnitActiveSec` is for consequent fires. I'll try to create a minimal docker image tomorrow to verify this though. If you have a Linux system feel free to test it.

Reply:
You were right, I've tested it and it doesn't work like that.

Reply:
To be honest, I'm not exactly sure how we could implement this for Linux

Reply:
@DenizUgur I'm sure @SMillerDev can help you figure that out

Reply:
It seems to me that https://github.com/Homebrew/homebrew-services/blob/cb8908eaf4ec3caf595b66cd5bd6c6511d63a431/lib/service/services_cli.rb#L252-L255 should check for `@run_at_load` and if it's false, not run `systemctl start` since in my understanding that immediately runs the service.

Reply:
I did this change but this makes `IMMEDIATE` run_type sorta obsolete, doesn't it?

Reply:
It does, but we need something to be the default. I'm not too attached to it 😁

Reply:
Yeah makes sense. Let's leave it like this then